### ISSUE TITLE
[TC] Only one NPC needed to summon Bots with no spawned Bots

### ISSUE THREAD

**Issue by Krutok** [trickerer/Trinity-Bots#761]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that my issue may get **closed without notice** if I *intentionally* remove or skip any mandatory\* field
- [X] This is NOT a bug/crash report
- [X] I've searched the bugtracker for similar questions/requests **including closed ones**

### Description

I would like to just place one NPC and then the player can summon the bots they need via a gossip menu. The general limit would be the bots available in the database and for players you could specify in the config how many bots of a class the player can have, e.g. a maximum of only 2 death knights or also generally specify how many bots the player is allowed to have and only in a raid Allow the Raidlead bots to spawn or release.

In other words, you have to let ALL the npcs spawn somewhere so that the player can choose the bots. Example: you want to set up bots for 2 40 raids and 25 raids, then you have to let a lot of bots spawn somewhere and that can also cause lags on the server.

If something like this already exists, please link to it.

I know there is already an NPC like this, but that refers to bots that have already spawned. What I would like to see is that it is no longer necessary to have all the bots spawn beforehand.

### TrinityCore or AzerothCore

TrinityCore

**Comment by trickerer** [comment]
It has been discussed already: #724. Also #601. As well as all the problems that come with this approach.
I may optimize server load like in #742, #579, but I'm not changing NPCBots concept. Bots are pre-spawned, period.

<details><summary>Spoiler</summary>
<p>

I have an idea to make bots only load their home grid when activated, like hired through bot giver npc, in some cases this could help a lot I think. This also would change an optimal bots spawn approach depending on the server.
</p>
</details>
